components/common-go/grpc/metrics.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2020 TypeFox GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package grpc
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// NewUnaryCallMetricsInterceptor creates a new unary interceptor that provides call counts
+// and error rate metrics.
+func NewUnaryCallMetricsInterceptor(reg prometheus.Registerer) (grpc.UnaryServerInterceptor, error) {
+	const (
+		labelMethod = "method"
+		labelCode   = "code"
+	)
+	callCountVec := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "grpc_call_total",
+		Help: "total call per function",
+	}, []string{labelMethod})
```

Comment:
Please correct me if I'm wrong, I'm not really familiar with gRPC.

Can we discover if a gRPC call failed by its status code? i.e. If we add `labelCode` to `gitpod_grpc_call_total`, would the two promQLs represent the same thing?

```
gitpod_grpc_error_total
```
```
gitpod_grpc_call_total{code!="200"}
```

If that is correct, I would suggest adding the code label to `gitpod_grpc_call_total`. We would reduce the number of metrics exposed and it would be possible to calculate the ratio of grpc calls per status code :wink:

Reply:
Good point - I've folded them both into the `gitpod_grpc_call_total` metric.